documentation/config_file_auth.md

```diff
@@ -9,142 +9,168 @@ canonical: "/puppetserver/latest/config_file_auth.html"
 
-~~~
-allow-unauthenticated: true
-~~~
-
-> **Note:** The authentication method introduced in Puppet Server 2.2 does not support the `allow_ip` or `deny_ip` parameters used in the deprecated `auth.conf` file.
+    If the request's authenticated name matches the parameter's value, Puppet Server allows it.
+* **`deny`**: This parameter can take the same types of values as the `allow` parameter, but refuses the request if the authenticated name matches --- even if the rule contains an `allow` value that also matches.
 
-In a deprecated `auth.conf` rule, the `auth` parameter specifies whether a rule applies only to authenticated clients (`on`; that is, those that provide a client certificate), only to unauthenticated clients (`off`), or to both (`any`).
+#### `sort-order`
 
-For example, the following deprecated `auth.conf` rule matches all clients, including those that do not have to be authenticated:
+After each rule's `match-request` section, the required `sort-order` parameter sets the order in which Puppet Server evaluates the rule by prioritizing it on a numeric value between 1 and 399 (to be evaluated before default Puppet rules) or 601 to 998 (to be evaluated after Puppet), with lower-numbered values evaluated first. Puppet Server secondarily sorts rules lexicographically by the `name` string value's Unicode code points.
 
-~~~
-auth: any
+~~~ hocon
+sort-order: 1
 ~~~
 
-> **Note:** In the HOCON Puppet Server authentication method, there is no directly equivalent behavior to the deprecated `auth` parameter's `on` value.
+#### `name`
 
-### `allow-header-cert-info`
+After each rule's `match-request` section, this required parameter's unique string value identifies the rule to Puppet Server. The `name` value is also written to server logs and error responses returned to unauthorized clients.
 
-> **Note:** If you've enabled the new authentication method introduced in Puppet Server 2.2, Puppet Server ignores the setting of the same name in [`master.conf`](./config_file_master.html) in favor of this setting in the new `auth.conf` file. If you use the deprecated authentication method and legacy [Puppet `auth.conf`][] file, you must instead configure this setting in `master.conf`.
+~~~ hocon
+name: "my path"
+~~~
 
-This setting determines whether to enable [external SSL termination](./external_ssl_termination.markdown) on all HTTP endpoints that Puppet Server handles, including those served by the "master" service, the certificate authority API, and the Puppet Admin API. It also controls how Puppet Server derives the user's identity for authorization purposes.
+> **Note:** If multiple rules have the same `name` value, Puppet Server will fail to launch.
 
-If this setting is enabled, Puppet Server ignores any presented certificate and relies completely on header data to authorize requests. **This is very insecure; do not do this unless you've secured your network to prevent _any_ untrusted access to Puppet Server.**
+#### `allow-unauthenticated`
```

Comment:
Would it be better to eliminate this section, in favor of the `allow, allow-unauthenticated, and deny` section?  Maybe the Notes about not supporting `allow_ip` and `deny_ip` and the `auth: on` parameter could instead be pulled up into the `allow-unauthenticated` bullet under the `allow, allow-unauthenticated, and deny` section?